.changeset/old-otters-arrive.md

```diff
@@ -0,0 +1,154 @@
+---
+'@commercetools-frontend/application-components': minor
+'@commercetools-website/custom-applications': minor
+'@commercetools-website/components-playground': minor
+---
+
+Add three new layout components: `<InfoDetailPage>`, `<FormDetailPage>` and `<CustomFormDetailPage>`.
+
+These components are similar to the `<InfoModalPage>`, `<FormModalPage>` and `<CustomFormModalPage>` respectively but they are not rendered as modals.
+
+# Usage
+
+The detail pages are supposed to be used as a direct child of one of the main pages.
+The layout of those pages can be recognized by the gray background header and the white content background. A back link in the header section of each of the pages is optional.
+
+## InfoDetailPage
+
+Form Detail pages are controlled components used to render a page with detailed data.
+
+```jsx
+import { useHistory } from 'react-router-dom';
+import { InfoDetailPage } from '@commercetools-frontend/application-components';
+import Text from '@commercetools-uikit/text';
+const DetailPage = () => {
+  const history = useHistory();
+  return (
+    <InfoDetailPage
+      title="Detail page"
+      showPageTopBar
+      onPreviousPathClick={() => history.push('/starting-page')}
+      previousPathLabel="Go back"
+    >
+      <Text.Body>{'Lorem ipsum ...'}</Text.Body>
+    </InfoDetailPage>
+  );
+};
+```
+
+## FormDetailPage
+
+Form Detail pages are controlled components used to render a form with predefined control elements (primary and secondary button).
+
+```jsx
+import { useHistory } from 'react-router-dom';
+import { useFormik } from 'formik';
+import TextField from '@commercetools-uikit/text-field';
+import TextInput from '@commercetools-uikit/text-input';
+import { FormDetailPage } from '@commercetools-frontend/application-components';
+
+const AccountPage = () => {
+  const history = useHistory();
+  const formik = useFormik({
+    initialValues: {
+      email: '',
+    },
+    validate: (formikValues) => {
+      if (TextInput.isEmpty(formikValues.email)) {
+        return { email: { missing: true } };
+      }
+      return {};
+    },
+    onSubmit: async (formikValues) => {
+      alert(`email: ${formikValues.email}`);
+      // Do something async
+    },
+  });
+  return (
+    <FormDetailPage
+      title="Manage your account"
+      showPageTopBar
+      onPreviousPathClick={() => history.push('/starting-page')}
+      isPrimaryButtonDisabled={formik.isSubmitting}
+      onSecondaryButtonClick={() => {
+        formik.resetForm();
+      }}
+      onPrimaryButtonClick={formik.handleSubmit}
+    >
+      <TextField
+        name="email"
+        title="Email"
+        isRequired={true}
+        value={formik.values.email}
+        errors={formik.errors.email}
+        touched={formik.touched.email}
+        onChange={formik.handleChange}
+        onBlur={formik.handleBlur}
+      />
+    </FormDetailPage>
```

Comment:
For both `FormDetailPage` and `CustomFormDetailPage` I'm missing the `form` html tag wrapping the input fields.

Personally, I would expect those components to internally add that element but, if we don't want that solution, at least I find it necessary to add it into the examples (both in the changelog and in the docs).

If we include the form tag in our components, perhaps we would also want to link the primary button to it using the form attribute ([HTML attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes); [form attribute explained](https://www.impressivewebs.com/html5-form-attribute/)). Although this can be tricky with the current implementation.

Reply:
I pushed a branch (**cm-details-pages**) with an implementation idea (not a working solution) of what I meant with my previous comment in case we want to discuss adding the form tag to the new form components.

Changes can be seen [here](https://github.com/commercetools/merchant-center-application-kit/commit/14bc21dfa08356ebd73a834af9e1e112b3280a24).

Reply:
FYI: I think using the `form` element is a valid point and we should discuss this separately, as we're also not really using it in many parts of the MC at the moment.

@CarlosCortizasCT Can you maybe open a discussion or issue to talk about it and how we plan to use it? We should also check with the other teams if there are other concerns about having it set by default in the components.

Thanks

Reply:
I'd also value this as I know many locations inside the MC where it's not used. Let's discuss it separately. I think we can either scan the next child and if it's a `form` not wrap it or have a `disableWrapByForm` prop which is `false` by default but can be disabled for instance. Similar to Next.js `Link` migration strategy which has a `oldBehaviour` prop.